test/uk/gov/hmrc/agentservicesaccount/forms/UpdateMoneyLaunderingSupervisionFormSpec.scala

```diff
@@ -0,0 +1,174 @@
+    }
+    s"error when $endDateMonth is invalid" in {
+      val params = Map(
+        bodyField -> "AABBCC",
+        numberField -> "11223344",
+
+        endDateDay -> local_Valid_date_stub.getDayOfMonth.toString,
+        endDateMonth -> "333",
+        endDateYear -> local_Valid_date_stub.getYear.toString
+      )
+      val validatedForm = UpdateMoneyLaunderingSupervisionForm.form.bind(params)
+      validatedForm.hasErrors shouldBe true
+      validatedForm.errors.size shouldBe 1
+      validatedForm.error(endDateField).get.message shouldBe "error.updateMoneyLaunderingSupervisory.date.invalid" //"month" check msg
+    }
+    s"error when $endDateYear is invalid" in {
+      val params = Map(
+        bodyField -> "AABBCC",
+        numberField -> "11223344",
+
+        endDateDay -> local_Valid_date_stub.getDayOfMonth.toString,
+        endDateMonth -> local_Valid_date_stub.getMonthValue.toString,
+        endDateYear -> "###"
+
+      )
+      val validatedForm = UpdateMoneyLaunderingSupervisionForm.form.bind(params)
+      validatedForm.hasErrors shouldBe true
+      validatedForm.errors.size shouldBe 1
+      validatedForm.error(endDateField).get.message shouldBe "error.updateMoneyLaunderingSupervisory.date.invalid" // "year" check msg
+    }
+    s"error when $endDateField passes the regex format however is an invalid date" in{
+      val params = Map(
+        bodyField -> "Blah alkfh",
+        numberField -> "1122334455",
+
+        endDateDay -> "31",
+        endDateMonth -> "02",
+        endDateYear -> local_Valid_date_stub.getYear.toString
+      )
+      val validatedForm = UpdateMoneyLaunderingSupervisionForm.form.bind(params)
+        validatedForm.error(endDateField).get.message shouldBe "error.updateMoneyLaunderingSupervisory.date.invalid"
+    }
+    s"error when $endDateField range isn't within 13Months of today's date" in{
+      val params = Map(
+        bodyField -> "Blah alkfh",
+        numberField -> "1122334455",
+
+        endDateDay -> local_future_date_stub.getDayOfMonth.toString,
+        endDateMonth -> local_future_date_stub.getMonthValue.toString,
+        endDateYear -> local_Valid_date_stub.getYear.toString
+      )
+      val validatedForm = UpdateMoneyLaunderingSupervisionForm.form.bind(params)
+
+        validatedForm.error(endDateField).get.message shouldBe "error.updateMoneyLaunderingSupervisory.date.past"
+    }
+    s"error when $endDateField is before today's date" in{
+      val params = Map(
+        bodyField -> "Blah alkfh",
+        numberField -> "1122334455",
+
+        endDateDay -> local_past_date_stub.getDayOfMonth.toString,
+        endDateMonth -> local_past_date_stub.getMonthValue.toString,
+        endDateYear -> local_past_date_stub.getYear.toString,
+      )
+      val validatedForm = UpdateMoneyLaunderingSupervisionForm.form.bind(params)
+        validatedForm.error(endDateField).get.message shouldBe "error.updateMoneyLaunderingSupervisory.date.past"
+    }
+  }
+}
```

Comment:
some good test coverage here for dates, but there's still probably some more scenarios to test like inputting any of day/month/year as 0 or maybe trying a '-1', might be worth reaching out to the QA specialists in the team to see if you can come up with some test scenarios together to put into these tests